### DISCUSSION TITLE
Unable to understand how priceFeed.version() works

### DISCUSSION THREAD

**Discussion by jasjyotsinghjaswal** [smartcontractkit/full-blockchain-solidity-course-py#131] [General]
1. At 2:47:00 We implemented the interface and alternatively copied the interface code **AggregatorV3Interface**, regardless there was no implementation for version method in the interface, so how are we still able to make the function calls . 

2. Also can you explain the difference between interface and abstract contract in Solidity

**Comment by PatrickAlphaC** [comment]
1. There is a function for version actually! https://github.com/smartcontractkit/chainlink/blob/8c1e22e02fd218d155a6868ee74a0e602ec58003/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol#L20

Just keep in mind, an interface doesn't need to define what a function does, it just needs to define what the function looks like, aka the function name, input parameters and return types. 

2. Abstract contracts are contracts with at least 1 function undefined, interfaces have all the functions undefined.
